content/doc/developer/plugin-development/injecting-environment-variables.adoc

```diff
@@ -0,0 +1,66 @@
+---
+layout: developerguide
+title: How to inject environment variables
+---
+
+Injecting some env variables allows to communicate steps with shell scripts and another
+steps. To achieve it, just follow this guide.
+We will add `EXTERNAL_SYSTEM_ID` environment variable in Pipeline compatible way.
+
+### Using implementation of SimpleBuildStep
+
+To be compatible both with old-fashion Freestyle jobs and brand new Pipeline, your step should
+implement `SimpleBuildStep` interface, which provides `perform(Run<?,?>, FilePath, Launcher, TaskListener)` method.
+
+First of all we should create specific implementation of `hudson.model.Action`. This will allow
+us to attach some information to the `Run` instance.
+
+Our implementation should store all required data for populating environment variable.
+
+[source,java]
+----
+@AllArgsConstructor <1>
+@Getter
+public class ExternalSystemLaunchedAction implements Action {
```

Comment:
you can replace `@AllArgsConstructor`, `@Getter` and `private` with a single annotation `@Value`

Reply:
I know, but this adds `toString`, `equals` and `hashcode` what is not my goal